include/deal.II/sundials/n_vector.templates.h

```diff
@@ -635,6 +639,24 @@ SUNDIALS::internal::NVectorOperations::weighted_rms_norm(N_Vector x, N_Vector w)
 
 
 
+template <typename VectorType>
+realtype
+SUNDIALS::internal::NVectorOperations::weighted_rms_norm_mask(N_Vector x,
+                                                              N_Vector w,
+                                                              N_Vector mask)
+{
+  // TODO copy can be avoided by a custom kernel
+  VectorType tmp         = *unwrap_nvector_const<VectorType>(x);
+  auto *     w_dealii    = unwrap_nvector_const<VectorType>(w);
+  auto *     mask_dealii = unwrap_nvector_const<VectorType>(mask);
+  const auto n           = tmp.size();
+  tmp.scale(*w_dealii);
+  tmp.scale(*mask_dealii);
+  return tmp.l2_norm() / std::sqrt(n);
```

Comment:
This fails for an empty vector x.

Reply:
Just like the other one. Sundials should throw in that case.